scripts/ci/images/ci_run_docker_tests.sh

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env bash
```

Comment:
I think if we are getting ride of bash we shoudl get rid of it entirely.
I think this could be very easily (and better) written as pure-stdib code that will prepare and activate the env with dependencies stored as `docker_test/requirements.txt' . In our case it would be:
`pytest, pytest-xdist, click` (the last one if we convert the script above to use click). 

This could be an 'entrypoint' to run tests - both from CI (installing such a small set of requirements can be done in CI and even cached easily) as well for a "fresh" user who does not have the venv installed - adding pytest-xdist automatically to such venv and running tests with it is good reason to have such auto-installation.

Reply:
This file is based on [ci_run_kubernetes_tests.sh](https://github.com/apache/airflow/blob/dddfedfdb1463f9e263dcc4570ff0fcee56b7c51/scripts/ci/kubernetes/ci_run_kubernetes_tests.sh), so rewriting it to Python is an extra effort which I agree we should make, but would prefer to do it in a separate contribution.

Reply:
Yeah but duplicating that will make it more difficult to remove in the future. I'd prefer to only add python code and remove bash code since we are going to get rid of bash. 

Seems like in this case we are trading one bash for another. Not a good idea.

Reply:
And you have a chance to establish a new "pattern" here.

Reply:
My main concern is that this script imports file `scripts/ci/libraries/_script_init.sh`, which initializes a lot of environment variables. If I wanted to write this file in Python I would have to find a way to access the following variables without using Bash: `BUILD_CACHE_DIR`, `AIRFLOW_PIP_VERSION`, `WHEEL_VERSION`, `CONSTRAINTS_GITHUB_REPOSITORY`, `DEFAULT_CONSTRAINTS_BRANCH`, `GITHUB_REGISTRY_PULL_IMAGE_TAG`.  I agree that duplicating code is not always the best idea, but generalizing this file and file `ci_run_kubernetes_tests.sh` in Bash would make it very complicated.

Reply:
Not really:

1) BUILD_CACHE_DIR is always `AIRFLOW_SOURCES/.build` - the ENV var is only defined once to use it in many scripts and do not make a typo, but this could be easily hardcoded in the python script and we can generalize it later when we change the approach in similar way to "Chart unit tests" and "Helm tests' (whcih we will be able to next)

2) pip install --upgrade pytest pytest-xdist - does not neeed constraints (that leaves most of the variables out). There is no reason why "environment" for "docker_tests" should be the same as Airifow one - we do not even install (and have no intentions to install) airflow here

3) similarly there is no reason to use the same pip and wheel versions - those are really needed to install airflow (which actually happens when we run Helm tests because we are using imports from airflow there). Just to install pytest and pytest-xdist we can use "current PIP and no wheel" - we do not even need to reinstall them

So what I am saying - this case and environment is far simpler, so it could be initialized in way simpler way - that's also one reason we should not carry all the "Breeze" baggage here - it's simply not needed here.

Reply:
And BTW. 

I really like the idea of small, dedicated venvs for different tasks - separate venv for "dev", separate for "docker_tests", separate for "chart/tests" - especially if we add some scripting that will create them when needed automtically. So we should reapply this to other cases in the future.

Reply:
I propose a solution later.